pandas/_libs/lib.pyx

```diff
@@ -1219,6 +1219,35 @@ cdef bint c_is_list_like(object obj, bint allow_sets) except -1:
     )
 
 
+def is_cursor(obj: object) -> bool:
```

Comment:
You can just move this to the sql.py file itself; I don't think this would be used elsewhere

Reply:
The intent is that it would be used by the `DataFrame` constructor after seeing if the `data` is list-like (see `pandas/core/frame.py` diff). I realize that is a bit invasive for such a special case, but it's the place where it is most generally useful for this use-case.